### PR TITLE
JAMES-3091 mailboxes get pojo

### PR DESCRIPTION
**Pull request by Arsnael**

### CODE REVIEW

File: server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/mail/MailboxGet.scala [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.jmap.mail
+
+import eu.timepit.refined.types.string.NonEmptyString
+import org.apache.james.jmap.model.Id.Id
+import org.apache.james.jmap.model.State.State
+
+case class AccountId(value: Id)
```
Comment (chibenwa) [inline server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/mail/MailboxGet.scala:26]: Use Session.AccountId

File: server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/mail/MailboxGet.scala [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+case class Ids(value: List[Id])
```
Comment (chibenwa) [inline server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/mail/MailboxGet.scala:28]: We should enforce somehow that these ids represent MailboxIds

File: server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/mail/MailboxGet.scala [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+case class Properties(value: List[NonEmptyString])
+
+case class MailboxGetRequest(accountId: AccountId,
+                             ids: Ids,
+                             properties: Properties)
+
+case class NotFound(value: List[Id])
```
Comment (chibenwa) [inline server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/mail/MailboxGet.scala:36]: Idem it is MailboxId, not just Id

**Comment by Arsnael** [comment]
Tests are :green_apple: Unexpected crash at the deployment step

test this please

File: server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/mail/MailboxGet.scala [hunk 4]
```diff
@@ -30,7 +30,7 @@ case class Properties(value: List[NonEmptyString])
 
 case class MailboxGetRequest(accountId: AccountId,
                              ids: Ids,
-                             properties: Properties)
+                             properties: Option[Properties])
```
Comment (Arsnael) [inline server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/mail/MailboxGet.scala:33]: `properties` is not mandatory actually, after reading again the specs

File: server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/mail/MailboxGet.scala [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+import org.apache.james.jmap.model.AccountId
+import org.apache.james.jmap.model.State.State
+import org.apache.james.mailbox.model.MailboxId
+
+case class Ids(value: List[MailboxId])
+
+case class Properties(value: List[NonEmptyString])
+
+case class MailboxGetRequest(accountId: AccountId,
+                             ids: Ids,
```
Comment (mbaechler) [inline server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/mail/MailboxGet.scala:32]: is empty list allowed?

Comment (Arsnael) [reply server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/mail/MailboxGet.scala:32]: From the specs:
```
ids: Id[]|null The ids of the Foo objects to return. If null, then all records of the data type are returned, if this is supported for that data type and the number of records does not exceed the maxObjectsInGet limit.
```

They don't really say about an empty list, so I guess it is allowed..? I'm not sure if we should treat it as the null case or as if there is a list but it's empty, then we don't return any mailboxes

WDYT?

Comment (Arsnael) [reply server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/mail/MailboxGet.scala:32]: Ah no wait... A bit further in the specs, for the `list` argument in response:

> list: Foo[] An array of the Foo objects requested. This is the empty array if no objects were found or if the ids argument passed in was also an empty array. The results MAY be in a different order to the ids in the request arguments. If an identical id is included more than once in the request, the server MUST only include it once in either the list or the notFound argument of the response.

So if the `ids` argument is an empty array, then we don't return any mailboxes

**Comment by Arsnael** [comment]
Merged

**Pull request closed without merging**